modules_integration_test.go

```diff
@@ -0,0 +1,430 @@
+package goja_test // this is on purpose in a separate package
+
+import (
+	"fmt"
+	"io/fs"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"testing/fstest"
+	"time"
+
+	"github.com/dop251/goja"
+)
+
+type simpleComboResolver struct {
+	mu           sync.Mutex
+	cache        map[string]cacheElement
+	reverseCache map[goja.ModuleRecord]string
+	fs           fs.FS
+	custom       func(interface{}, string) (goja.ModuleRecord, error)
+}
+type cacheElement struct {
+	m   goja.ModuleRecord
+	err error
+}
+
+func newSimpleComboResolver() *simpleComboResolver {
+	return &simpleComboResolver{cache: make(map[string]cacheElement), reverseCache: make(map[goja.ModuleRecord]string)}
+}
```

Comment:
(this likely can be in a separate PR later)

I have made some simple implementations for what is needed to get modules working as part of the modules "integration" tests - that also are specifically not in the same package but `goja_test` so I enforce using the package as user. 

Some of those types and helpers are fairly basic and useful and might be a good idea to be separated in a package that can be used by users that will just do something fairly standard.

I guess goja_nodejs might be a good place. We might even try to make what I am making for k6 - being able to `import` commonJS and `require` ESM modules. This likely though will be more involved and for ... mostly historical reasons - k6 hasn't been using that project and likely we will never move to it. So it might not get as good of a usage from us as the rest of the code here.